### PR TITLE
Stat verbs

### PR DESCRIPTION
**Pull request by LethalGhost**
Add basic statverbs mechanic.
Statverb - kind ability to apply stat to any object.
When someone examine object it will see cilable list of applicable stats which he can use.

### CODE REVIEW

**Comment by SpadesSlick** [comment]
Provide description, please.

**Comment by mistyLuminescence** [comment]
Is this still being worked on?

**Comment by drexample** [comment]
@LethalGhost Please, provide more description and applications.

File: code/game/atoms.dm [hunk 1]
```diff
@@ -262,6 +262,8 @@ its easier to just keep the beam vertical.
 	else
 		user.visible_message("<font size=1>[user.name] looks at [src].</font>", "\icon[src] This is [full_name] [suffix]")
 
+	user << show_stat_verbs() //rewrite to show_stat_verbs(user)?
```
Comment (drexample) [inline code/game/atoms.dm]: goonchat this pls

File: code/controllers/subsystems/staverbs.dm [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	var/required_stat = STAT_MEC
+	var/base_range = RANGE_ADJACENT	//maximum distance or RANGE_ADJACENT
+	var/minimal_stat = 0
+
+
+/datum/statverb/proc/try_action(mob/living/user, atom/target, saved_name = "target")
+	if(!istype(user))
+		return
+	if(!target in view(user))
+		user << SPAN_WARNING("You're too far from [saved_name]")
```
Comment (drexample) [inline code/controllers/subsystems/staverbs.dm]: goonchat

File: code/controllers/subsystems/staverbs.dm [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+		return FALSE
+
+	if(base_range == RANGE_ADJACENT)
+		if(!target.Adjacent(user))
+			user << SPAN_WARNING("You should be adjacent to [target]")
```
Comment (drexample) [inline code/controllers/subsystems/staverbs.dm]: goonchat

File: code/controllers/subsystems/staverbs.dm [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+			return FALSE
+	else
+		if(get_dist(user, target) > base_range)
+			user << SPAN_WARNING("You're too far from [target]")
```
Comment (drexample) [inline code/controllers/subsystems/staverbs.dm]: goonchat

File: code/controllers/subsystems/staverbs.dm [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+			return FALSE
+
+	if(user.stats.getStat(required_stat) < minimal_stat)
+		user << SPAN_WARNING("You're not skilled enought in [required_stat]")
```
Comment (drexample) [inline code/controllers/subsystems/staverbs.dm]: goonchat

File: code/game/atoms.dm (hunk 1 above)
Comment (drexample) [inline code/game/atoms.dm]: Fortn... goonchat

**Comment by drexample** [comment]
@LethalGhost pls, update the PR, it's hanging here for two months.

**Pull request merged**